src/gtktest.rs

```diff
@@ -3,295 +3,189 @@
-
-        let ret = dialog.run();
-
-        dialog.destroy();
-
-        entry_clone.set_text(&format!("Clicked {}", ret));
-    }));
-
-    // use a plain function instead of a closure
-    button_about.connect_clicked(about_clicked);
-
-    button_font.connect_clicked(clone!(window => move |_| {
-        let dialog = gtk::FontChooserDialog::new(Some("Font chooser test"), Some(&window));
+        );
+    }
 
-        dialog.run();
-        dialog.destroy();
-    }));
+    fn about_clicked(button: &Button, builder: &Builder) {
+        let dialog: AboutDialog = builder.get_object("dialog").unwrap();
+        if let Some(window) = button.get_toplevel().and_then(|w| w.downcast::<Window>().ok()) {
+            dialog.set_transient_for(Some(&window));
+        }
 
-    button_recent.connect_clicked(clone!(window => move |_| {
-        let dialog = gtk::RecentChooserDialog::new(Some("Recent chooser test"), Some(&window));
-        dialog.add_buttons(&[
-            ("Ok", gtk::ResponseType::Ok.into()),
-            ("Cancel", gtk::ResponseType::Cancel.into())
-        ]);
+        println!("Authors: {:?}", dialog.get_authors());
+        println!("Artists: {:?}", dialog.get_artists());
+        println!("Documenters: {:?}", dialog.get_documenters());
 
         dialog.run();
         dialog.destroy();
-    }));
-
-    file_button.connect_clicked(clone!(window => move |_| {
-        //entry.set_text("Clicked!");
-        let dialog = gtk::FileChooserDialog::new(Some("Choose a file"), Some(&window),
-            gtk::FileChooserAction::Open);
-        dialog.add_buttons(&[
-            ("Open", gtk::ResponseType::Ok.into()),
-            ("Cancel", gtk::ResponseType::Cancel.into())
-        ]);
-
-        dialog.set_select_multiple(true);
-        dialog.run();
-        let files = dialog.get_filenames();
-        dialog.destroy();
-
-        println!("Files: {:?}", files);
-    }));
+    }
 
-    app_button.connect_clicked(clone!(window => move |_| {
-        //entry.set_text("Clicked!");
-        let dialog = gtk::AppChooserDialog::new_for_content_type(Some(&window), gtk::DIALOG_MODAL,
-            "sh");
+    pub fn sub_main() {
+        if gtk::init().is_err() {
+            println!("Failed to initialize GTK.");
+            return;
+        }
+        println!("Major: {}, Minor: {}", gtk::get_major_version(), gtk::get_minor_version());
+        let glade_src = include_str!("gtktest.glade");
+        let builder = Builder::new_from_string(glade_src);
+
+        let window: Window = builder.get_object("window").unwrap();
+        let entry: Entry = builder.get_object("entry").unwrap();
+        let spinner: Spinner = builder.get_object("spinner").unwrap();
+        spinner.start();
+
+        /*info_bar.connect(signals::Response::new(|response_id| {
+            info_bar.hide()
+        }));*/ //TODO: Why does this not work?
+
+        let scale: Scale = builder.get_object("scale").unwrap();
+        scale.set_range(0., 100.);
```

Comment:
> It's not a big deal, none of us know GTK very well. It's puzzling that you'd go on not knowing and offload the googling to someone else.

That's quite offending that you might I just let things like this in hope someone will do it for me. I let it like this because I wasn't able to find the way to do it.